src/endpoints/ECatCourse/index.ts

```diff
@@ -744,6 +756,384 @@ class ECatCourse extends EndpointCategory {
+    const assignmentMap: { [k: number]: number } = {};
+    // iterate through each source assignment to determine the mapping
+    sourceAssignments.forEach((sourceAssignment) => {
+      const destinationAssignment = destinationAssignments.find((assignment) => {
+        return assignment.name === sourceAssignment.name;
+      });
+      if (destinationAssignment) {
+        assignmentMap[sourceAssignment.id] = destinationAssignment.id;
+      } else {
+        throw new CACCLError({
+          message: 'Could not find a migrated assignment in the destination course.',
+          code: ErrorCode.MigrationIssue,
+        });
+      }
+    });
+
+    // iterate through each assignment group in the source course and
+    // create the same assignment group in the destination course
+    const sourceAssignmentGroups = await this.api.course.assignmentGroup.list({
+      courseId: sourceCourseId,
+    });
+    const sourceAssignmentGroupIds = sourceAssignmentGroups.map((group) => { return group.id; });
+    for (let i = 0; i < sourceAssignmentGroupIds.length; i++) {
+      const sourceId = sourceAssignmentGroupIds[i];
+      const sourceAssignmentGroup = await this.api.course.assignmentGroup.get({
+        assignmentGroupId: sourceId,
+        courseId: sourceCourseId,
+      });
+      // check if apply_assignment_group_weights is true in the source course
+      const sourceCourse = await this.visitEndpoint({
+        path: `${API_PREFIX}/courses/${sourceCourseId}`,
+        action: 'get source course',
+        method: 'GET',
+      });
+      const applyAssignmentGroupWeights = sourceCourse.apply_assignment_group_weights;
+      // TODO: check if the assignment group name already exists in the destination course,
+      // in which we case we do not create a new assignment group
+      // instead, get the id of this matching assignment group and update weights if needed
+      // and also add this assignment group to the map
+      let destinationAssignmentGroup;
+      if (applyAssignmentGroupWeights) {
+        destinationAssignmentGroup = await this.api.course.assignmentGroup.create({
+          courseId: destinationCourseId,
+          name: sourceAssignmentGroup.name,
+          weight: sourceAssignmentGroup.group_weight,
+        });
+      } else {
+        destinationAssignmentGroup = await this.api.course.assignmentGroup.create({
+          courseId: destinationCourseId,
+          name: sourceAssignmentGroup.name,
+        });
+      }
+      // set apply_assignment_group_weights to true/false in the destination course
+      await this.visitEndpoint({
+        path: `${API_PREFIX}/courses/${destinationCourseId}`,
+        action: 'set apply_assignment_group_weights to true',
+        method: 'PUT',
+        params: {
+          course: {
+            apply_assignment_group_weights: applyAssignmentGroupWeights,
+          },
+        },
+      });
+
+      // add the mapping to the map
```

Comment:
Could we change this to be more descriptive?

`// add assignment group mapping`